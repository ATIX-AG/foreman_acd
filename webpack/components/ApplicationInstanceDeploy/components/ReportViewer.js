import React from 'react';
import PropTypes from 'prop-types';

const ReportViewer= ({
  hidden,
  report,
}) =>{
  if (hidden) {
    return null;
  }

  if (report === undefined) {
    return null;
  }

  return report.map((task, id) => (
    <span key={"report_task_"+id} className="report_task">{task.name}</span>
  ));
};

ReportViewer.propTypes = {
  hidden: PropTypes.bool,
  report: PropTypes.array,
};

export default ReportViewer;
