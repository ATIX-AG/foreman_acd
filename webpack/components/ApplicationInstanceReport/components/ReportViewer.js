import React from 'react';
import PropTypes from 'prop-types';

const ReportViewer= ({
  hidden,
  report,
}) =>{
  if (hidden) {
    return null;
  }

  if (report === undefined) {
    return null;
  }

  return report.map((task, id) => (
    <div key={"report_task_"+id} className="report_task">{task.name} -- {task.status}</div>
  ));
};

ReportViewer.propTypes = {
  hidden: PropTypes.bool,
  report: PropTypes.array,
};

export default ReportViewer;
