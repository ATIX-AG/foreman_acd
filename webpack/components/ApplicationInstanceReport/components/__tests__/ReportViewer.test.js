import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import ReportViewer from '../ReportViewer';

const noop = () => {};

const data = [
  { name: 'report 1', status: 'done' },
  { name: 'report 2', status: 'pending' },
];

const fixtures = {
  'should render the report viewer': {
    hidden: false,
    report: data,
  },
  'should render the hidden report viewer': {
    hidden: true,
    report: data,
  },
};

describe('ReportViewer', () =>
  testComponentSnapshotsWithFixtures(ReportViewer, fixtures));
