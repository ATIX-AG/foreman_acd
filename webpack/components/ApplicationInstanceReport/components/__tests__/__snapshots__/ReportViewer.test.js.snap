// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportViewer should render the hidden report viewer 1`] = `""`;

exports[`ReportViewer should render the report viewer 1`] = `
Array [
  <div
    className="report_task"
    key="report_task_0"
  >
    report 1
     -- 
    done
  </div>,
  <div
    className="report_task"
    key="report_task_1"
  >
    report 2
     -- 
    pending
  </div>,
]
`;
